packages/fgtclb/academic-persons/Classes/Upgrades/ListTypeToCTypeUpgradeWizard.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FGTCLB\AcademicPersons\Upgrades;
+
+use Doctrine\DBAL\Schema\Column;
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Install\Attribute\UpgradeWizard;
+use TYPO3\CMS\Install\Updates\DatabaseUpdatedPrerequisite;
+use TYPO3\CMS\Install\Updates\UpgradeWizardInterface;
+
+#[UpgradeWizard('academicPersons_MigrateListTypeToCTypeContentElements')]
+final class ListTypeToCTypeUpgradeWizard implements UpgradeWizardInterface
+{
+    public function __construct(
+        private readonly ConnectionPool $connectionPool,
+    ) {}
+
+    /**
+     * This must return an array containing the "list_type" to "CType" mapping.
+     * The array key is the exact value of corresponding "tt_content.list_type" DB records,
+     * the array value is the new "CType" value.
+     * not plugin
+     *
+     *  Example:
+     *
+     *  [
+     *      'pi_plugin1' => 'pi_plugin1',
+     *      'pi_plugin2' => 'new_content_element',
+     *  ]
+     *
+     * Note that string keys with integer values like '4' will be treated as INT by
+     * PHP internally, which is why string-casting is performed later on.
+     * @see https://3v4l.org/JNPfU
+     *
+     * @return array<string|int, string>
+     */
+    protected function getListTypeToCTypeMapping(): array
+    {
+        return [
+            'academicpersons_detail' => 'academicpersons_detail',
+            'academicpersons_list' => 'academicpersons_list',
+            'academicpersons_listanddetail' => 'academicpersons_listanddetail',
+            'academicpersons_selectedcontracts' => 'academicpersons_selectedcontracts',
+            'academicpersons_selectedprofiles' => 'academicpersons_selectedprofiles',
```

Comment:
@sbuerk Sorry, I just realised now there is a sixth plugin that it was not being taken into account in this list. The next line should be added:

```
'academicpersons_card' => 'academicpersons_card',
```

Reply:
> @sbuerk Sorry, I just realised now there is a sixth plugin that it was not being taken into account in this list. The next line should be added:
> 
> ```
> 'academicpersons_card' => 'academicpersons_card',
> ```

Nope, because `academicpersons_card` was never a list_type plugin, it was a CType plugin right from the start: https://github.com/fgtclb/academic-extensions/commit/c1e6c25c8ff380221c9b960f73758218ae0cdc7b

So no need to migrate it. If project "changed that in a violating way" it's up to them to provide an upgrade wizard.